### ISSUE TITLE
Please do not encourage non-idiomatic Windows-like installation schemes on Linux

### ISSUE THREAD

**Issue by hasufell** [haskell/ghcup-hs#335]
In GitLab by @navid-zamani on Mar 7, 2022, 02:06

$HOME is not for software installation but for user data.  
Software should integrate with or be installed via the package manager.  
On Linux/Unix, files of a package are installed in their correct respective directories.  
It follows a philosophy of sharing.

GHCup suggests installing GHC and everything into $HOME/.ghcup instead of standard location.  
GHCup circumvents the package manager. And hence,  
GHCup follows a philosophy of erecting walls and duplicating data (and inflating the home directory).

Disadvantageous behavior like that was introduced to Linux by developers from a closed-source / corporate / Windows background, who kept treating Linux like it was just another Windows. Namely in the form of containers, and with Steam.  
GHCup should of course not follow these tendencies, or, if the developers prefer this kind of style, should stay on an OS that actually works that way, instead of ruining something they clearly refused to understand.  
At least until, in 40 years, Windows too, is doomed to reimplement Unix/Plan9/Inferno, badly. ;)

**TL;DR: Just write normal packages. It’s much easier than this ugly hack.** The hack’s only necessary on operating systems with prehistoric package managers that fail at dependency resolution because they can’t handle multiple version of the same package. So if anything, it’s only an offering for, frankly, legacy OSes.
